### DISCUSSION TITLE
[5430 AC] 런타임 에러(NumberFormat)가 발생하는데 이유를 모르겠습니다.

### DISCUSSION THREAD

**Discussion by yukyeongmin** [sds-2021-summer-algorithm/java#80] [Q&A] [answered]
[https://help.acmicpc.net/judge/rte/NumberFormat](url)
vscode에서 돌렸을때는 이런 에러가 안나는데 이유가 뭘까요..?
체점하면 50퍼센트에서 위의 에러가 납니다.

저는 포인터같은 2개의 변수 start, end이용해서 삭제나 뒤집기를 구현 하지 않고 
마지막에 출력만 맞추는 식으로 구현해보려고 했어요

  `

        public class Main {
            static int M;
            static int [] data;
            static int start;
            static int end;
            public static void main(String[] args) throws IOException{
                System.setIn(new FileInputStream("P5430/input.txt"));
                BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
                BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
                int N = Integer.parseInt(br.readLine());
        
                for(int i = 0 ; i<N; i++){
                    String command = br.readLine();
                    M = Integer.parseInt(br.readLine()); // 데이터 길이
                    if(M==0) {
                        bw.write("error\n");
                        continue;
                    }
                    data = new int[M+1];
                    String line = br.readLine();
                    line = line.substring(1, line.length()-1);
                    String[] nums = line.split(",");
                    for(int j = 0 ; j<nums.length ; j++){
                        data[j] = Integer.parseInt(nums[j]);                
                    }
        
                    start = 0; // 데이터의 시작
                    end = M; // 다음데이터가 들어갈 자리
                    boolean error = false;
                    for(int j = 0 ; j<command.length(); j++){
                        char c = command.charAt(j);
                        if(c=='D'){
                            if(start==end) {
                                error = true;
                                break;
                            }else if(start<end){
                                start++;
                            }else{
                                start--;
                            }
                        }else if(c=='R'){
                            if(j!=command.length()-1 && command.charAt(j+1)=='R'){
                                j++;
                                continue;
                            }
                            if(start>end){
                                int temp = start;
                                start = end+1;
                                end = temp+1;
                            }else{
                                int temp = start;
                                start = end-1;
                                end = temp-1;
                            }
                        }
                    }
        
                    if(error) bw.write("error\n");
                    else{
                        bw.write("[");
                        if(start<end){
                            for(int j = start; j<end; j++){
                                bw.write(Integer.toString(data[j]));
                                if(j!=end-1) bw.write(",");
                            }
                        }else{
                            for(int j = start; j>end; j--){
                                bw.write(Integer.toString(data[j]));
                                if(j!=end+1) bw.write(",");
                            }
                        }
        
                        bw.write("]\n");
                    }
                }
                bw.flush();
                bw.close();
                br.close();
            }
        }
`

**Comment by dls4585** [comment, marked as answer]
R일 때 start와 end를 swap 해주는 과정에서 +1 -1 을 해주는 이유가 있나요 ??

**Reply by yukyeongmin** [reply]
start == end인 지점을 데이터수가 0인걸로 보려고
start는 시작위치고 end는 마지막 데이터 위치 +1이라서 그 형태를 유지해 주려고 했어요!

**Reply by yukyeongmin** [reply]
인찬님 코드보고 boolean flag만드는게 낫겠다 싶어서 R부분을 그냥 뒤집힘 여부로 처리해버렸는데 그래도 같은 오류가 나네요ㅠㅠ

`
              else if(c=='R'){
                                  if(j!=command.length()-1 && command.charAt(j+1)=='R'){
                                      j++;
                                      continue;
                                  }
                                  if(flag) flag = false;
                                  else flag = true;
                              }
`

**Reply by yukyeongmin** [reply]
앗 인찬님이 댓글달았다가 삭제하신거 알림 메일로 확인했는데 그게 정답이었어요!! 그부분 수정하니까 통과했습니다.
역시 소중한 답변이 맞았어요!!!
